python3/agents/totoro_agent/brain.py

```diff
@@ -2,26 +2,23 @@
 Tracks the game state and decides on next strategy to execute
 """
 
-from ..shared.trackers import BombTracker, EnemyTracker, MapTracker, PickupTracker
+from ..shared.trackers import FinalsTracker
 from ..shared.utils.benchmark import Benchmark
 
 
 class Brain:
     def __init__(self):
         self.benchmark = Benchmark()
-        self.bomb_tracker = BombTracker()
-        self.enemy_tracker = EnemyTracker()
-        self.map_tracker = MapTracker()
-        self.pickup_tracker = PickupTracker()
+        self.finals_tracker = FinalsTracker()
 
     def get_next_strategy(self, game_state) -> str:
         self.benchmark.start('tracker')
         """Conditionals to decide which strategy to execute. Returns string"""
-        self.enemy_tracker.update(game_state)
-        self.bomb_tracker.update(game_state)
-        self.map_tracker.update(game_state)
-        self.pickup_tracker.update(game_state)
+        self.finals_tracker.update(game_state)
         self.benchmark.end('tracker')
+        self.finals_tracker.update_path(game_state)
```

Comment:
Maybe do this update to places where relevant

Reply:
E.g. we first start out with the general update,

then we update path so that new gamestate properties are accessible to the next checks, etc